internal/engine/entities/entity_event.go

```diff
@@ -347,6 +347,33 @@ func (eiw *EntityInfoWrapper) GetEntityDBIDs() (repoID uuid.NullUUID, artifactID
 	return repoID, artifactID, pullRequestID
 }
 
+// GetID returns the entity ID.
+func (eiw *EntityInfoWrapper) GetID() (uuid.UUID, error) {
+	if eiw == nil {
+		return uuid.Nil, fmt.Errorf("no entity info wrapper")
+	}
+
+	id, ok := eiw.getIDForEntityType(eiw.Type)
+	if ok {
+		return id, nil
+	}
+
+	return uuid.Nil, fmt.Errorf("no entity ID found")
+}
+
+func (eiw *EntityInfoWrapper) getIDForEntityType(t minderv1.Entity) (uuid.UUID, bool) {
```

Comment:
I guess you could make the methods simpler by making `getIDForEntityType` accept a nil receiver and return uuid.Nil in that case as well as not return the second bool from `getIDForEntityType` but just uuid.Nil. But it's a matter of taste, both are fine.